src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -1455,11 +1455,11 @@ private void UpdateWindowProperties(WindowProperties newProperties, bool forceCh
                     style &= ~WindowStyles.WS_MINIMIZEBOX;
 
                 if (newProperties.IsMaximizable || (newProperties.WindowState == WindowState.Maximized && newProperties.IsResizable))
-                    style |= WindowStyles.WS_MAXIMIZEBOX;
+                    style |= WindowStyles.WS_MAXIMIZEBOX | WindowStyles.WS_SYSMENU;
```

Comment:
This is a no-op in all cases, at this point `WS_SYSMENU` is already included. See the initial definition of the `style` variable which has `WS_OVERLAPPEDWINDOW` (which itself includes `WS_SYSMENU`).